File: Problem 14 - Longest Common Prefix/code.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        commonPrefix = ""
        shortestLength = len(strs[0])
        
        for i in range(shortestLength):
            currentCharacter = strs[0][i]
            for j in range(len(strs)):
                if strs[j][i] == currentCharacter:
                    if j == len(strs)-1:
                        commonPrefix = commonPrefix + currentCharacter
                else:
                    return commonPrefix
    
    # Case 1 
    # Input: strs = ["flower","flow","flight"]
    # Expected Output: "fl"
    print(longestCommonPrefix("", ["flower","flow","flight"]))

    # Case 2
    # Input: strs = ["dog","racecar","car"]
    # Expected Output: ""
    print(longestCommonPrefix("", ["dog","racecar","car"]))
            

File: Problem 14 - Longest Common Prefix/code.js
var longestCommonPrefix = function(strs) {
    var commonPrefix = ""
    
    for(let i = 0; i < strs[0].length; i++) {
        var currentCharacter = strs[0][i]
        for(let j = 0; j < strs.length; j++) {
            if(strs[j][i] == currentCharacter){
                if(j == strs.length-1) {
                    commonPrefix = commonPrefix + currentCharacter
                }
            }
            else {
                return commonPrefix
            }
        }
    }
    
    return commonPrefix
};

// Case 1 
// Input: strs = ["flower","flow","flight"]
// Expected Output: "fl"
console.log(longestCommonPrefix(["flower","flow","flight"]))

// Case 2
// Input: strs = ["dog","racecar","car"]
// Expected Output: ""
console.log(longestCommonPrefix(["dog","racecar","car"]))